src/main/java/com/aws/greengrass/device/iot/IotAuthClient.java

```diff
@@ -62,19 +65,23 @@
                 VerifyClientDeviceIdentityResponse response = RetryUtils.runWithRetry(SERVICE_EXCEPTION_RETRY_CONFIG,
                         () -> clientFactory.getGreengrassV2DataClient().verifyClientDeviceIdentity(request),
                         "verify-client-device-identity", logger);
-                return Optional.of(response.clientDeviceCertificateId());
+                String certificateId = response.clientDeviceCertificateId();
+                logger.atInfo().kv(CERTIFICATE_PEM, certificatePem)
```

Comment:
Not sure if we need to log the client cert PEM at Info level. We should be careful about this.

Reply:
client certificate should not contain sensitive info, such as private key. But probably not very useful to log certificate here either, will remove.